neqo-http3/src/features/extended_connect/webtransport_streams.rs

```diff
@@ -145,6 +152,11 @@ impl SendStream for WebTransportSendStream {
     fn send(&mut self, conn: &mut Connection) -> Res<()> {
         if let WebTransportSenderStreamState::SendingInit { ref mut buf, fin } = self.state {
             let sent = conn.stream_send(self.stream_id, &buf[..])?;
+            // We only want to count bytes when the state is SendingData.
+            // However, the underlying connection is not aware of the state
+            // here. That's why we need to count the bytes sent before
+            // SendingData state.
+            self.bytes_non_app_data += sent as u64;
```

Comment:
Given that we can calculate this value based on the encoded size of the session ID (varint length of the session ID / 4), do we really need to track this in this way?

Reply:
Sorry, I am not sure I really understand this.
@martinthomson  Could you explain a bit more? Thanks.

Reply:
It looks like you are trying to calculate the data sent on a stream, excluding the stream header (in this case, that is the identifier we send to indicate that the stream belongs to a particular webtransport session).  We track the total number of bytes sent on a stream, but subtract the size of this header from the value that we report.

But we know what that overhead is and can calculate it when we report the webtransport-specific stats.  So rather than keep a variable that only ever holds a value of 1, we can calculate the value instead (with `Encoder::varint_len()` and the value from here, which we should know).